includes/class-sensei-course-structure.php

```diff
@@ -129,29 +136,38 @@ public function filter_no_cache_where( $where ) {
 	 *
 	 * @see Sensei_Course_Structure::prepare_lesson()
 	 *
-	 * @param WP_Term      $module_term        Module term.
-	 * @param array|string $lesson_post_status Lesson post status(es).
+	 * @param WP_Term      $module_term               Module term.
+	 * @param array|string $lesson_post_status        Lesson post status(es).
+	 * @param boolean      $add_teacher_name_property Teacher's name will be added to the module as a property.
 	 *
 	 * @return array {
 	 *     An array which has module information.
 	 *
 	 *     @type string $type        The type of the array which is 'module'.
 	 *     @type int    $id          The module term id.
 	 *     @type string $title       The module name.
+	 *     @type string $teacher     Name of the teacher of this module, gets populated only in admin panel for admins and if the teacher is not admin.
 	 *     @type string $description The module description.
 	 *     @type array  $lessons     An array of the module lessons. See Sensei_Course_Structure::prepare_lesson().
 	 * }
 	 */
-	private function prepare_module( WP_Term $module_term, $lesson_post_status ) : array {
+	private function prepare_module( WP_Term $module_term, $lesson_post_status, $add_teacher_name_property ) : array {
 		$lessons = $this->get_module_lessons( $module_term->term_id, $lesson_post_status );
-		$module  = [
+
+		$module = [
 			'type'        => 'module',
 			'id'          => $module_term->term_id,
 			'title'       => $module_term->name,
 			'description' => $module_term->description,
+			'teacher'     => '',
 			'lessons'     => [],
 		];
 
+		if ( $add_teacher_name_property ) {
+			$author            = Sensei_Core_Modules::get_term_author( $module_term->slug );
+			$module['teacher'] = user_can( $author, 'manage_options' ) ? '' : $author->display_name;
+		}
```

Comment:
What do you think about always adding the teacher here to make things simpler and not adding specific logic to the method? This way you won't need the `$add_teacher_name_property` argument. Example:

```php
$author = Sensei_Core_Modules::get_term_author( $module_term->slug );
if ( $author && in_array( 'teacher', $author->roles, true ) ) {
	$module['teacher'] = $author->display_name;
}
```

Reply:
Another option here would be to create a new filter and call it in the controller to append the teacher.